### PR TITLE
tiltfile: on manifest creation, pull matching yaml l out of globalYAML and add to manifest [ch681]

### PR DESCRIPTION
**Pull request by maiamcc**
Hello @jazzdan, @nicks,

Please review the following commits I made in branch maiamcc/parse-global-yaml:

9574cf099829caded9a8feefedcd161a57f22a5b (2018-10-30 18:06:06 -0400)
tiltfile: on manifest creation, pull matching yaml l out of globalYAML and add to manifest [ch681]

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: internal/engine/upper.go [hunk 1]
```diff
@@ -92,13 +92,15 @@ func (u Upper) RunHud(ctx context.Context) error {
...
+	globalYAML model.YAMLManifest, watchMounts bool) error {
+	span, ctx := opentracing.StartSpanFromContext(ctx, "Up")
 	defer span.Finish()
 
 	u.store.Dispatch(InitAction{
-		WatchMounts: watchMounts,
-		Manifests:   manifests,
+		WatchMounts:        watchMounts,
+		Manifests:          manifests,
+		GlobalYAMLManifest: globalYAML,
```
Comment (jazzdan) [inline internal/engine/upper.go]: Wait- I'm confused: in this code already in master? https://github.com/windmilleng/tilt/blob/6da59ebe249922c9758dc3d40cd59bba23eaaac4/internal/engine/upper.go#L103

**Comment by nicks** [comment]
ya, can you sync the pr so we can see what's different?

**Comment by maiamcc** [comment]
lol sorry. Better?

File: internal/yaml/utils.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package yaml
+
+import (
+	"fmt"
+	"strings"
+)
+
+const separator = "---"
+
+func ConcatYaml(yaml ...string) string {
```
Comment (nicks) [inline internal/yaml/utils.go]: Yaml -> YAML
per golang naming conventions

File: internal/yaml/utils.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	}
+
+	result := yaml[0]
+	for _, y := range yaml[1:] {
+		result = concatYaml(result, y)
+	}
+	return result
+}
+
+func concatYaml(y1, y2 string) string {
```
Comment (nicks) [inline internal/yaml/utils.go]: i wonder if we should eventually store yaml in the data model as a []string, without separators. this is definitely crying out for a more structured representation

**Pull request merged**